### PR TITLE
Added listing of all the commands.

### PR DESCRIPTION
**Pull request by wqyeo**
Based on #3 

`help` just simply lists out all the commands in an embed.<br>The user can navigate through the different pages using the LeftArrow and RightArrow emoji.

*Alternatively*, the user can also filter the search by `help (filterHere)`.<br>In-code, it just uses `string.Contains` on the `Summary`, `Remarks` and `Name` attribute if applicable.

Pages are separated based on modules rather than length.

>The bot will just say a warning if the filter failed to find any respective commands.
The bot will not show the respective emojis if the commands can fit into a page.

### Code Design
**For handling Message Reactions**

`CommandHandler` makes the respective static `MessageReactionHandler` function subscribe to the respective `Client` object reaction event-handlers.

To make an object have a callback when a reaction-related event occurs to a message, let it implement `IMessageReactionHandle` and call `AddMessageReactionHandle` from `MessageReactionHandler`, with the input being the respective object.

Just remember to call `RemoveMessageReactionHandle` after you are done, with the input being the respective object.

### CODE REVIEW

File: Cake.Core/Discord/Modules/HelpModule/HelpModule.cs [hunk 1]
```diff
@@ -25,22 +25,32 @@ public HelpModule(HelpService service, CommandService commandService)
...
+            {
+                await newMessage.ModifyAsync(msg => msg.Content = "There is no matching command based on the search filter!");
+            }
+            else if (helpPages.Count == 1)
+            {
+                await newMessage.ModifyAsync(msg => { msg.Embed = helpPages[0].Build(); msg.Content = string.Empty; });
+            }
+            else
+            {
+                await newMessage.ModifyAsync(msg => { msg.Embed = helpPages[0].Build(); msg.Content = string.Empty; });
```
Comment (Darius-Wattimena) [inline Cake.Core/Discord/Modules/HelpModule/HelpModule.cs]: Here might be a better place to do a check if the count != 1

This way you can remove the else if you have right now

File: Cake.Core/Discord/Handlers/CommandHandler.cs [hunk 2]
```diff
@@ -52,16 +56,18 @@ private async Task PrefixCommandAsync(ShardedCommandContext context, int argPos
...
             {
+                // TODO: UNCOMMENT THESE:
                 //Get or generation of user.
-                var user = await Database.Queries.UserQueries.FindOrCreateUser(context.User.Id);
-                var guild = await Database.Queries.GuildQueries.FindOrCreateGuild(context.Guild.Id);
+                //var user = await Database.Queries.UserQueries.FindOrCreateUser(context.User.Id);
+                //var guild = await Database.Queries.GuildQueries.FindOrCreateGuild(context.Guild.Id);
 
-                if (context.Message.HasCharPrefix(Convert.ToChar(guild.Prefix), ref argPos))
+                if (context.Message.HasCharPrefix(Convert.ToChar(">"/*guild.Prefix*/), ref argPos))
```
Comment (Darius-Wattimena) [inline Cake.Core/Discord/Handlers/CommandHandler.cs]: Set back stuff like how it was first

**Review by Darius-Wattimena — CHANGES_REQUESTED** [review]

**Comment by wqyeo** [comment]
> Here might be a better place to do a check if the count != 1
> 
>This way you can remove the else if you have right now

👍

**Comment by CakeAndBanana** [comment]
Hello, Sorry for the late response (we were on vacation in Germany).

I was testing out the helpbook function and if you go from last page to next page, it goes to the second last page. Also can you add page numbers at the bottom as a footer.

Can you add a spacer before the description text of the command?

**Comment by wqyeo** [comment]
@CakeAndBanana  👍 
Hopefully I did not misinterpret your meaning of `add a spacer before the description text of the command?`

**Comment by CakeAndBanana** [comment]
Uhm, The page number is going backwards. When I'm going forward in the pages it goes from 3 to 2 and when I'm going backwards it's going from 2 to 3. Ignore the spacer part, I can't explain it well enough 😄

**Comment by wqyeo** [comment]
@CakeAndBanana 

> Uhm, The page number is going backwards. When I'm going forward in the pages it goes from 3 to 2 and when I'm going backwards it's going from 2 to 3.

Fixed it; 
P.S, I was thinking about books that flipped Left-to-Right 😅

**Comment by CakeAndBanana** [comment]
Thanks for this PR, Everything looks fine.
I will push this to master :)

**Pull request merged**